pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -67,6 +127,9 @@ func ApplyParameters(ctx context.Context, spec *v1.TaskSpec, tr *v1.TaskRun, def
 					arrayReplacements[fmt.Sprintf(pattern, p.Name)] = p.Default.ArrayVal
 				}
 			case v1.ParamTypeObject:
+				for _, pattern := range paramPatterns {
+					objectReplacements[fmt.Sprintf(pattern, p.Name)] = p.Default.ObjectVal
+				}
```

Comment:
Yes, like I mentioned in my previous comment, it is needed now inside a Task. For inlined Steps, it was not needed before. Therefore, I don't think we can split this into another PR since it would be meaningless.

Reply:
Sounds good 👍🏾